sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1857,25 +1873,11 @@ def _provision(self,
                 head_ip = config_dict['head_ip']
                 ip_list = [head_ip]
```

Comment:
In many other places, we will check if the `head_ip` is set in the `handle` to determine whether the cluster is up, e.g. https://github.com/skypilot-org/skypilot/blob/6a48a127e73dc0cdf452c2db3e8eca2de19c0fbb/sky/core.py#L340

and https://github.com/skypilot-org/skypilot/blob/6a48a127e73dc0cdf452c2db3e8eca2de19c0fbb/sky/core.py#L611-L612

We need to cache the head_ip into the handle after provisioning.

Reply:
Doesn't this mean that we cannot remove `head_ip` and replace it with as an `@property` variable?

Reply:
Could you explain more?

Previously, in L1877, we have the `head_ip`for creating the `handle`, but now we removed it. I mean why cannot us have the `ip_list` cached in the handle after it is provisioned?

Reply:
I see, I didn't realize you were referring to the specific case of a single-node cluster. I believe we cannot use the optimization here anymore because we need both internal and external IP of the head node for single-node clusters, but parsing the Ray output can only give us the external IP. I've updated the code so that `get_stable_cluster_ips` is always called after provisioning, and this will cache the ip list in the handle.